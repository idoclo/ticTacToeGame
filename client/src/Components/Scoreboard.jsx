import React, { Component } from 'react';
import { Header, Table } from 'semantic-ui-react';


class Scoreboard extends Component {
  constructor() {
    super();
    this.state = { topThreePlayers: [] }
  }

  componentWillMount() {
    fetch('/players/', { method: 'GET' })
    .then(playersInfo => playersInfo.json())
    .then(resJSON => {
      console.log('info on players', resJSON);
      const playersInfo = resJSON.slice();
      let first;
      let second;
      let third;
      let tempIndex;
      // First place
      playersInfo.forEach((player, index) => {
        if (!first || player.score > first.score) {
          first = player;
          tempIndex = index;
        }
      });
      // Slice out first player
      playersInfo.splice(tempIndex, 1);
      // Second place
      playersInfo.forEach((player, index) => {
        if (!second || player.score > second.score) {
          second = player;
          tempIndex = index;
        }
      });
      // Slice out second player
      playersInfo.splice(tempIndex, 1);
      // Third place
      playersInfo.forEach((player, index) => {
        if (!third || player.score > third.score) {
          third = player;
          tempIndex = index;
        }
      });

      this.setState({ topThreePlayers: [
          { name: first.username, score: first.score },
          { name: second.username, score: second.score },
          { name: third.username, score: third.score }
        ]
      })
    })
    .catch(err => {
      console.error('Unable to get all players info', err);
    });
  }

  componentDidMount() {
    const { topThreePlayers } = this.state;
    console.log('playa', topThreePlayers);
  }

  render() {
    const { topThreePlayers } = this.state;
    return (
      <div>
        <Header size="small">
          Top three players
        </Header>
        <Table celled>
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell>Player</Table.HeaderCell>
              <Table.HeaderCell>Score</Table.HeaderCell>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            {topThreePlayers.map(player => (
              <Table.Row>
                <Table.Cell>{player.name}</Table.Cell>
                <Table.Cell>{player.score}</Table.Cell>
              </Table.Row>
            ))}
          </Table.Body>
        </Table>
      </div>
    );
  }
};

export default Scoreboard;