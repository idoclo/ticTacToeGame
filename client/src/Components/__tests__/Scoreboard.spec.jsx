import React from 'react';
import { shallow } from 'enzyme';
import "isomorphic-fetch";
import Scoreboard from '../Scoreboard';


describe('Scoreboard', () => {
  const shallowComponent = shallow(<Scoreboard gameOn={false} />);

  test('Scoreboard renders correctly', () => {
    expect(shallowComponent).toMatchSnapshot();
  });
});