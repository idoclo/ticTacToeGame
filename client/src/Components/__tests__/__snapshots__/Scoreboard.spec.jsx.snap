// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard Scoreboard renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scoreboard
    gameOn={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          size="small"
        >
          Top three players
        </Header>,
        <Table
          as="table"
          celled={true}
        >
          <TableHeader
            as="thead"
          >
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableHeaderCell
                as="th"
              >
                Player
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Score
              </TableHeaderCell>
            </TableRow>
          </TableHeader>
          <TableBody
            as="tbody"
          />
        </Table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Top three players",
          "size": "small",
        },
        "ref": null,
        "rendered": "Top three players",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "table",
          "celled": true,
          "children": Array [
            <TableHeader
              as="thead"
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Player
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Score
                </TableHeaderCell>
              </TableRow>
            </TableHeader>,
            <TableBody
              as="tbody"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "thead",
              "children": <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Player
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Score
                </TableHeaderCell>
              </TableRow>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tr",
                "cellAs": "td",
                "children": Array [
                  <TableHeaderCell
                    as="th"
                  >
                    Player
                  </TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
                  >
                    Score
                  </TableHeaderCell>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Player",
                  },
                  "ref": null,
                  "rendered": "Player",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Score",
                  },
                  "ref": null,
                  "rendered": "Score",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "tbody",
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            size="small"
          >
            Top three players
          </Header>,
          <Table
            as="table"
            celled={true}
          >
            <TableHeader
              as="thead"
            >
              <TableRow
                as="tr"
                cellAs="td"
              >
                <TableHeaderCell
                  as="th"
                >
                  Player
                </TableHeaderCell>
                <TableHeaderCell
                  as="th"
                >
                  Score
                </TableHeaderCell>
              </TableRow>
            </TableHeader>
            <TableBody
              as="tbody"
            />
          </Table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Top three players",
            "size": "small",
          },
          "ref": null,
          "rendered": "Top three players",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "table",
            "celled": true,
            "children": Array [
              <TableHeader
                as="thead"
              >
                <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableHeaderCell
                    as="th"
                  >
                    Player
                  </TableHeaderCell>
                  <TableHeaderCell
                    as="th"
                  >
                    Score
                  </TableHeaderCell>
                </TableRow>
              </TableHeader>,
              <TableBody
                as="tbody"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "thead",
                "children": <TableRow
                  as="tr"
                  cellAs="td"
                >
                  <TableHeaderCell
                    as="th"
                  >
                    Player
                  </TableHeaderCell>
                  <TableHeaderCell
                    as="th"
                  >
                    Score
                  </TableHeaderCell>
                </TableRow>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "tr",
                  "cellAs": "td",
                  "children": Array [
                    <TableHeaderCell
                      as="th"
                    >
                      Player
                    </TableHeaderCell>,
                    <TableHeaderCell
                      as="th"
                    >
                      Score
                    </TableHeaderCell>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "th",
                      "children": "Player",
                    },
                    "ref": null,
                    "rendered": "Player",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "th",
                      "children": "Score",
                    },
                    "ref": null,
                    "rendered": "Score",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tbody",
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
